spec/Makefile

```diff
@@ -23,26 +23,31 @@ report-regression:
 #
 
 # Spec heaps.
-HEAPS += ASpec ASpecDoc ExecSpec DSpec CKernel CSpec TakeGrant ASepSpec
+HEAPS += ASpec ExecSpec DSpec CKernel CSpec TakeGrant ASepSpec
 
 # Additional dependencies
 
 # CKernel uses the `machinety=machine_state` option for `install_C_file`,
 # and therefore depends on `design-spec`.
 CKernel CSpec: c-kernel design-spec
 
-# NOTE: The abstract spec imports Events from Haskell hence the dependency
-ASpec ExecSpec DSpec : design-spec
+ExecSpec DSpec : design-spec
 
-ASPEC_DOC_GITREV_FILE=abstract/document/gitrev.tex
-ASPEC_DOC_VERSION_FILE=abstract/document/VERSION
+ASPEC_VERSION_FILE=abstract/document/VERSION
+ASPEC_GITREV_FILE=abstract/document/gitrev.tex
+GIT_ROOT_FILE=abstract/document/git-root.tex
 
-ASpecDoc: design-spec $(ASPEC_DOC_VERSION_FILE) $(ASPEC_DOC_GITREV_FILE)
+# NOTE: The abstract spec imports Events from Haskell hence the dependency
+ASpec: $(ASPEC_VERSION_FILE) $(GIT_ROOT_FILE) $(ASPEC_GITREV_FILE) design-spec
 
-$(ASPEC_DOC_VERSION_FILE): $(SEL4_VERSION)
+$(ASPEC_VERSION_FILE): $(SEL4_VERSION)
 	cp $< $@
 
-$(ASPEC_DOC_GITREV_FILE): .FORCE
+$(GIT_ROOT_FILE):
+	git rev-parse --show-toplevel > $@
+	perl -p -i -e "s/\n//" $@
+
+$(ASPEC_GITREV_FILE): .FORCE
 	git rev-parse --short=15 HEAD > $@ || echo unknown > $@
 	git diff --no-ext-diff --quiet || echo "*" >> $@
 	git diff --no-ext-diff --quiet --cached || echo "+" >> $@
```

Comment:
For anyone curious, see the comments @lsf37 added in `spec/abstract/document/root.tex`.